contracts/libraries/StringUtils.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {strings} from "./strings.sol";
+
+// Extracted from https://github.com/zkemail/email-wallet-sdk/blob/main/src/helpers/StringUtils.sol
+library StringUtils {
+    using strings for *;
+
+    /**
+     * @dev Converts a hexadecimal string to a bytes32 value.
+     * @param hexStr The hexadecimal string to convert.
+     * @return result A bytes32 value representing the converted hexadecimal string.
+     */
+    function hexToBytes32(string calldata hexStr) public pure returns (bytes32 result) {
+        require(hexStr.toSlice().startsWith("0x".toSlice()), "invalid hex prefix");
+        hexStr = hexStr[2:];
+        require(bytes(hexStr).length == 64, "invalid hex string length");
+        uint256[] memory ints = hex2Ints(hexStr);
+        uint256 sum = 0;
+        for (uint256 i = 0; i < 32; i++) {
+            sum = (256 * sum + ints[i]);
+        }
+        return bytes32(sum);
+    }
+
+    /**
+     * @dev Converts a hexadecimal string to a bytes.
+     * @param hexStr The hexadecimal string to convert. Must start with "0x" prefix.
+     * @return A bytes array representing the converted hexadecimal string.
+     */
+    function hexToBytes(string calldata hexStr) public pure returns (bytes memory) {
+        require(hexStr.toSlice().startsWith("0x".toSlice()), "invalid hex prefix");
+        string memory hexStrNoPrefix = hexStr[2:];
+        bytes memory hexBytes = bytes(hexStrNoPrefix);
+        require(hexBytes.length % 2 == 0, "invalid hex string length");
+
+        bytes memory result = new bytes(hexBytes.length / 2);
+
+        for (uint256 i = 0; i < hexBytes.length / 2; i++) {
+            result[i] = bytes1(
+                (hexChar2Int(hexBytes[2 * i]) << 4) + hexChar2Int(hexBytes[2 * i + 1])
```

Comment:
@aalimsahin out of interest, how come you went for this approach, instead of doing something similar to `hexToBytes32`? E.g. something like this:
```solidity
        uint256[] memory ints = hex2Ints(hexStrNoPrefix);
        uint256 length = bytes(hexStrNoPrefix).length / 2;
        bytes memory result = new bytes(length);
        for (uint256 i = 0; i < length; i++) {
            result[i] = bytes1(uint8(ints[i]));
        }
        return result;
```

Reply:
I just thought the version I added was more readable. I found the first version harder to understand. I preferred the second version because I found the binary to hexadecimal conversion easier to understand.

Reply:
Thanks, we will update to use your implementation